### DISCUSSION TITLE
Can I build Android C++ library so that I can use C++ api for Android Native C Development?

### DISCUSSION THREAD

**Discussion by xinyuliu006** [k2-fsa/sherpa-onnx#1319] [Q&A]
I followed [this](https://k2-fsa.github.io/sherpa/onnx/c-api/index.html) instruction and want to cross-compile this to an Android library. I want to use api of this project in android native C++ development. I tried these commands:
```
cmake \
  -DSHERPA_ONNX_ENABLE_C_API=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DBUILD_SHARED_LIBS=ON \
  -DCMAKE_INSTALL_PREFIX=/tmp/sherpa-onnx/shared \
  -DCMAKE_TOOLCHAIN_FILE=$NDK/build/cmake/android.toolchain.cmake \
  -DANDROID_ABI=x86_64  \
  -DANDROID_PLATFORM=latest  \
  -DCMAKE_C_FLAGS="-march=znver2 -mtune=znver2 -O3"  \
  ..
make
``` 
When "making", errors happend.
![image](https://github.com/user-attachments/assets/c6d0df33-b50c-498a-b517-07434426a433)
What should I do? Thank you for any help.

**Comment by csukuangfj** [comment]
What is the version of your sdk?

**Reply by xinyuliu006** [reply]
I only downloaded NDK, which is android-ndk-r27-linux.

**Reply by xinyuliu006** [reply]
The first error is:
```
ld.lld: error: unable to find library -lpthread
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [_deps/portaudio-build/CMakeFiles/portaudio.dir/build.make:289: lib/libsherpa-ncnn-portaudio.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:348: _deps/portaudio-build/CMakeFiles/portaudio.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
``` 
The second error is:
```
fatal error: error in backend: Cannot select: 0x5c9627585370: v32bf16 = insert_subvector 0x5c9627667720, 0x5c9627666bc0, Constant:i64<16>, ncnn-src/src/layer/x86/x86_usability.h:1326:18 @[ ncnn-src/src/layer/x86/cast_bf16.h:63:51 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ] ]
  0x5c9627667720: v32bf16 = insert_subvector undef:v32bf16, 0x5c9627666c30, Constant:i64<0>, ncnn-src/src/layer/x86/x86_usability.h:1326:18 @[ ncnn-src/src/layer/x86/cast_bf16.h:63:51 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ] ]
    0x5c96275858b0: v32bf16 = undef
    0x5c9627666c30: v16bf16 = X86ISD::CVTNEPS2BF16 nnan ninf nsz arcp contract afn reassoc 0x5c96276676b0, ncnn-src/src/layer/x86/x86_usability.h:1324:20 @[ ncnn-src/src/layer/x86/cast_bf16.h:63:51 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ] ]
      0x5c96276676b0: v16f32,ch = load<(load (s512) from %ir.59, align 1, !tbaa !1702, !noalias !1618)> 0x5c96274c76c0, 0x5c9627667aa0, undef:i64, ncnn-src/src/layer/x86/cast_bf16.h:63:71 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ]
        0x5c9627667aa0: i64,ch = CopyFromReg 0x5c96274c76c0, Register:i64 %17, ncnn-src/src/layer/x86/cast_bf16.h:63:71 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ]
          0x5c96279e0f80: i64 = Register %17
        0x5c96279e0650: i64 = undef
    0x5c96279e05e0: i64 = Constant<0>
  0x5c9627666bc0: v16bf16 = X86ISD::CVTNEPS2BF16 nnan ninf nsz arcp contract afn reassoc 0x5c9627585300, ncnn-src/src/layer/x86/x86_usability.h:1325:20 @[ ncnn-src/src/layer/x86/cast_bf16.h:63:51 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ] ]
    0x5c9627585300: v16f32,ch = load<(load (s512) from %ir.61, align 1, !tbaa !1702, !noalias !1618)> 0x5c96274c76c0, 0x5c9627667950, undef:i64, ncnn-src/src/layer/x86/cast_bf16.h:63:93 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ]
      0x5c9627667950: i64 = add 0x5c9627667aa0, Constant:i64<64>, ncnn-src/src/layer/x86/cast_bf16.h:63:93 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ]
        0x5c9627667aa0: i64,ch = CopyFromReg 0x5c96274c76c0, Register:i64 %17, ncnn-src/src/layer/x86/cast_bf16.h:63:71 @[ ncnn-src/src/layer/x86/cast_bf16.h:51:5 ]
          0x5c96279e0f80: i64 = Register %17
        0x5c96279e1060: i64 = Constant<64>
      0x5c96279e0650: i64 = undef
  0x5c96275856f0: i64 = Constant<16>
In function: _ZN4ncnnL21cast_fp32_to_bf16_sseERKNS_3MatERS0_RKNS_6OptionE.omp_outlined
PLEASE submit a bug report to https://github.com/android-ndk/ndk/issues and include the crash backtrace, preprocessed source, and associated run script.

```

**Reply by csukuangfj** [reply]
please don't use ndk 27


 there is a bug in NDK 27

**Reply by xinyuliu006** [reply]
Thank you, I changed the NDK, but it still has errors.
![image](https://github.com/user-attachments/assets/d6fb3841-015d-4fd3-82e6-4d1c39d721e4)

**Reply by csukuangfj** [reply]
what is the ndk version?


could you post all.the  logs?

**Reply by xinyuliu006** [reply]
Sorry for not offering all the details. The NDK I'm using now is **android-ndk-r26d-linux**
Here is the cmake log:
```
cmake   \
-DSHERPA_NCNN_ENABLE_C_API=ON   \
-DCMAKE_BUILD_TYPE=Release   \
-DBUILD_SHARED_LIBS=ON  \
 -DCMAKE_INSTALL_PREFIX=install \
-DCMAKE_TOOLCHAIN_FILE=/home/xinyu/android-ndk-r26d-linux/android-ndk-r26d/build/cmake/android.toolchain.cmake \
-DANDROID_ABI=x86_64 \
-DANDROID_PLATFORM=latest ..
``` 
<details><summary>Result</summary>
<p>

$ cmake   -DSHERPA_NCNN_ENABLE_C_API=ON   -DCMAKE_BUILD_TYPE=Release   -DBUILD_SHARED_LIBS=ON   -DCMAKE_INSTALL_PREFIX=install -DCMAKE_TOOLCHAIN_FILE=/home/xinyu/android-ndk-r26d-linux/android-ndk-r26d/build/cmake/android.toolchain.cmake -DANDROID_ABI=x86_64 -DANDROID_PLATFORM=latest ..
-- Using latest available ANDROID_PLATFORM: 34.
-- The C compiler identification is Clang 17.0.2
-- The CXX compiler identification is Clang 17.0.2
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/xinyu/android-ndk-r26d-linux/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/xinyu/android-ndk-r26d-linux/android-ndk-r26d/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Set SHERPA_NCNN_ENABLE_JNI to ON for Android
-- CMAKE_EXPORT_COMPILE_COMMANDS: ON
-- BUILD_SHARED_LIBS ON
-- SHERPA_NCNN_ENABLE_PYTHON OFF
-- SHERPA_NCNN_ENABLE_PORTAUDIO ON
-- SHERPA_NCNN_ENABLE_JNI ON
-- SHERPA_NCNN_ENABLE_BINARY ON
-- SHERPA_NCNN_ENABLE_TEST OFF
-- SHERPA_NCNN_ENABLE_C_API ON
-- SHERPA_NCNN_ENABLE_GENERATE_INT8_SCALE_TABLE ON
-- SHERPA_NCNN_ENABLE_FFMPEG_EXAMPLES OFF
-- SHERPA_NCNN_ENABLE_WASM OFF
-- SHERPA_NCNN_ENABLE_WASM_FOR_NODEJS OFF
-- CMAKE_BUILD_TYPE: Release
-- Looking for C++ include alsa/asoundlib.h
-- Looking for C++ include alsa/asoundlib.h - not found
CMake Warning at CMakeLists.txt:112 (message):
    Could not find alsa/asoundlib.h !
    We won't build sherpa-ncnn-alsa
    To fix that, please do:
      (1) sudo apt-get install alsa-utils libasound2-dev
      (2) rm -rf build
      (3) re-try
    


-- Downloading kaldi-native-fbank from https://github.com/csukuangfj/kaldi-native-fbank/archive/refs/tags/v1.18.7.tar.gz
-- kaldi-native-fbank is downloaded to /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/kaldi_native_fbank-src
-- kaldi-native-fbank's binary dir is /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/kaldi_native_fbank-build
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_EXPORT_COMPILE_COMMANDS: ON
-- BUILD_SHARED_LIBS: ON
-- KALDI_NATIVE_FBANK_BUILD_TESTS: OFF
-- KALDI_NATIVE_FBANK_BUILD_PYTHON: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- CMAKE_CXX_FLAGS: -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  
-- CMAKE_INSTALL_PREFIX: /home/xinyu/TTS/sherpa-ncnn/build-adb/install
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Looking for C++ include execinfo.h
-- Looking for C++ include execinfo.h - found
-- Disable building Python
-- Downloading ncnn from https://github.com/Tencent/ncnn/archive/964ed7a56a573c5046cc177a6cd95580ecddcddf.zip
-- ncnn is downloaded to /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src
-- ncnn's binary dir is /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-build
-- CMAKE_TOOLCHAIN_FILE = /home/xinyu/android-ndk-r26d-linux/android-ndk-r26d/build/cmake/android.toolchain.cmake
-- CMAKE_INSTALL_PREFIX = /home/xinyu/TTS/sherpa-ncnn/build-adb/install
-- NCNN_VERSION_STRING = 1.0.20240906
-- Performing Test NCNN_COMPILER_SUPPORT_GNU_INLINE_ASM
-- Performing Test NCNN_COMPILER_SUPPORT_GNU_INLINE_ASM - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_FMA
-- Performing Test NCNN_COMPILER_SUPPORT_X86_FMA - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_XOP
-- Performing Test NCNN_COMPILER_SUPPORT_X86_XOP - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_F16C
-- Performing Test NCNN_COMPILER_SUPPORT_X86_F16C - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX2
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX2 - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512 - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX_VNNI
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX_VNNI - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_VNNI
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_VNNI - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_BF16
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_BF16 - Success
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_FP16
-- Performing Test NCNN_COMPILER_SUPPORT_X86_AVX512_FP16 - Success
-- Target arch: x86 64bit
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Found OpenMP_C: -fopenmp=libomp (found version "5.0") 
-- Found OpenMP_CXX: -fopenmp=libomp (found version "5.0") 
-- Found OpenMP: TRUE (found version "5.0")  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Downloading portaudio from http://files.portaudio.com/archives/pa_stable_v190700_20210406.tgz
-- portaudio is downloaded to /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/portaudio-src
-- portaudio's binary dir is /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/portaudio-build
CMake Deprecation Warning at build-adb/_deps/portaudio-src/CMakeLists.txt:7 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (jack) does
  not match the name of the calling package (Jack).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  build-adb/_deps/portaudio-src/cmake_support/FindJack.cmake:34 (find_package_handle_standard_args)
  build-adb/_deps/portaudio-src/CMakeLists.txt:277 (FIND_PACKAGE)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find jack (missing: JACK_LIBRARY JACK_INCLUDE_DIR) 
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/xinyu/TTS/sherpa-ncnn/build-adb


</p>
</details> 

```
make -j 10
``` 


<details><summary>Result</summary>
<p>
[  0%] Built target ncnn-generate-spirv 
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/fftsg.cc.o
[  1%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_converters.c.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-fbank.cc.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-window.cc.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-functions.cc.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/mel-computations.cc.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/rfft.cc.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/online-feature.cc.o
[  2%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_allocation.c.o
[  2%] Generating source unaryop_x86_fma.h
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/portaudio-src/src/common/pa_converters.c:350:32: warning: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Wimplicit-const-int-float-conversion]
        double scaled = *src * 0x7FFFFFFF;
                             ~ ^~~~~~~~~~
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/portaudio-src/src/common/pa_converters.c:407:32: warning: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Wimplicit-const-int-float-conversion]
        double scaled = *src * 0x7FFFFFFF;
                             ~ ^~~~~~~~~~
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/portaudio-src/src/common/pa_converters.c:535:32: warning: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648 [-Wimplicit-const-int-float-conversion]
        double scaled = *src * 0x7FFFFFFF;
                             ~ ^~~~~~~~~~
[  2%] Generating source binaryop_x86_avx.cpp
[  2%] Generating source binaryop_x86_avx.h
[  2%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_cpuload.c.o


（略）


[ 14%] Generating source crop_x86_avx512.h
[ 14%] Generating source crop_x86_fma.cpp
[ 14%] Generating source crop_x86_fma.h
[ 15%] Generating source dequantize_x86_avx.cpp
[ 15%] Generating source dequantize_x86_avx.h
**ld.lld: error: unable to find library -lpthread
clang: error: linker command failed with exit code 1 (use -v to see invocation)**
[ 15%] Generating source dequantize_x86_avx512.cpp
make[2]: *** [_deps/portaudio-build/CMakeFiles/portaudio.dir/build.make:289: lib/libsherpa-ncnn-portaudio.so] Error 1
[ 15%] Generating source dequantize_x86_fma.cpp
make[1]: *** [CMakeFiles/Makefile2:348: _deps/portaudio-build/CMakeFiles/portaudio.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 15%] Generating source dequantize_x86_avx512.h
[ 16%] Generating source dequantize_x86_fma.h


（略）


[ 48%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/modelbin.cpp.o
[ 49%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/net.cpp.o
In file included from /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/cpu.cpp:123:
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/ruapu.h:117:31: warning: missing field '' initializer [-Wmissing-field-initializers]
    struct sigaction sa = { 0 };
                              ^
[ 49%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/option.cpp.o
[ 49%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/paramdict.cpp.o
[ 49%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/pipeline.cpp.o
[ 50%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/pipelinecache.cpp.o
[ 50%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/simpleocv.cpp.o
[ 50%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/simpleomp.cpp.o
[ 50%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/simplestl.cpp.o
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/net.cpp:2477:37: warning: unused parameter 'num_threads' [-Wunused-parameter]
void Extractor::set_num_threads(int num_threads)
                                    ^
[ 51%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/simplemath.cpp.o
1 warning generated.
[ 51%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/simplevk.cpp.o
cnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_avx512.cpp.o
[ 54%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_fma.cpp.o
[ 54%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_avx.cpp.o
[ 54%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_avx512vnni.cpp.o
1 warning generated.
[ 54%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_avxvnni.cpp.o
[ 55%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_avx2.cpp.o
In file included from /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-build/src/layer/x86/convolution_x86_avx512.cpp:48:
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/layer/x86/convolution_im2col_gemm_int8.h:4874:27: warning: explicitly assigning value of variable of type '__m512i' (vector of 8 'long long' values) to itself [-Wself-assign]
                    _sum0 = _sum0;
                    ~~~~~ ^ ~~~~~
[ 55%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/convolution_x86_xop.cpp.o
In file included from /home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/layer/x86/convolution_x86_avx512vnni.cpp:22:
/home/xinyu/TTS/sherpa-ncnn/build-adb/_deps/ncnn-src/src/layer/x86/convolution_im2col_gemm_int8.h:4874:27: warning: explicitly assigning value of variable of type '__m512i' (vector of 8 'long long' values) to itself [-Wself-assign]
                    _sum0 = _sum0;
                    ~~~~~ ^ ~~~~~
[ 55%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/crop.cpp.o
[ 55%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/crop_x86.cpp.o
[ 56%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/x86/crop_x86_avx512.cpp.o


（略）


[ 88%] Building CXX object _deps/ncnn-build/src/CMakeFiles/ncnn.dir/layer/glu.cpp.o
[ 88%] Linking CXX shared library ../../../lib/libncnn.so
[ 88%] Built target ncnn
make: *** [Makefile:136: all] Error 2



</p>
</details>

**Reply by csukuangfj** [reply]
Could you try the ndk version used by our building script?

**Reply by csukuangfj** [reply]
Remember to delete the build directory if you change the NDK version.

**Comment by csukuangfj** [comment]
Without additional information, we are unfortunately not sure how to resolve this discussion. We are therefore reluctantly going to close this discussion for now. Please don't hesitate to comment on the discussion if you have any more information for us; we will reopen it right away!
